sqlglot/generator.py

```diff
@@ -2063,7 +2070,19 @@ def select_sql(self, expression: exp.Select) -> str:
             self.sql(expression, "into", comment=False),
             self.sql(expression, "from", comment=False),
         )
-        return self.prepend_ctes(expression, sql)
+
+        sql = self.prepend_ctes(expression, sql)
+
+        if self.SUPPORT_SELECT_INTO == False and into:
+            # Ensures that SELECT INTO TEMP UNLOGGED -> CREATE TEMPORARY TABLE
+            kind = (
+                " TEMPORARY"
+                if into.args.get("temporary")
+                else (" UNLOGGED" if into.args.get("unlogged") else "")
+            )
```

Comment:
Let's rename this to `table_kind` since `kind` is used above in a different context. Also, I'm not sure if `UNLOGGED` is correct here - have you checked whether dialects support the `CREATE UNLOGGED ... TABLE ...` syntax?

My guess is you need to just do `table_kind = " TEMPORARY" if into.args.get("temporary") else ""` here.

Reply:
Ok nevermind, I just verified that postgres supports the `CREATE UNLOGGED TABLE ... AS ...` syntax, so we can keep this logic.

Reply:
The ideal case actually would be to control whether or not to generate `UNLOGGED` with another flag, because that way we would avoid generating invalid CTAS statements for dialects that don't support unlogged tables.

So I'd rewrite this into something that looks like this:

```python
if into.args.get("temporary"):
    kind = " TEMPORARY"
elif self.SUPPORTS_UNLOGGED_TABLES and into.args.get("unlogged"):
    kind = " UNLOGGED"
else:
    kind = ""
```